src/components/Statuses/Statuses.tsx

```diff
@@ -23,10 +25,46 @@ const Statuses = () => {
     return <div className={styles.base}>Keine Daten vorhanden</div>;
   }
 
+  //helper function to get dateKey from status departureTime
+  const getDateKey = (status: Status) => {
+    const date = new Date(
+      status.train.origin.departureReal ?? status.train.origin.departure ?? ''
+    );
+    //remove time from date
+    date.setHours(0, 0, 0, 0);
+    return date;
+  };
+
+  //Group statuses by date
+  const groupedStatuses: { dateKey: Date; statuses: Status[] }[] = [];
+
+  statuses.forEach((status) => {
+    //get Date of status and check if it already exists in groupedStatuses
+    const dateKey = getDateKey(status);
+    const existingDate = groupedStatuses.find(
+      (group) => group.dateKey.toDateString() === dateKey.toDateString()
+    );
+    if (existingDate) {
+      existingDate.statuses.push(status);
+    } else {
+      groupedStatuses.push({
+        dateKey,
+        statuses: [status],
+      });
+    }
+  });
```

Comment:
I would recommend to use `reduce` and create a real map instead of an array of key/value items. That would also simplify the code a lot.

```suggestion
  const statusGroups = statuses.reduce((groups, status) => {
    const key = getDateKey(status);
    
    return { ...groups, [key]: [...groups[key] ?? [], status] };
  };
```

Reply:
Never heard of a .reduce method before. Will look into it, as it makes the code much more readable.